### PR TITLE
Implement derivatives for torch.remainder and torch.fmod wrt the second argument and update the docs

### PR DESCRIPTION
**Pull request by lezcano**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #67833
* #68188
* #67996
* #68523
* #68522
* #69909
* __->__ #69908

I also took this chance to clarify a bit the documentation of these
functions.

cc @brianjo @mruberry

Differential Revision: [D33774417](https://our.internmc.facebook.com/intern/diff/D33774417)

### CODE REVIEW

**Comment by Lezcano** [comment]
Tagged @albanD and @nikitaved for the derivative and @mruberry for the docs.

File: torch/_torch_docs.py [hunk 1]
```diff
@@ -8287,10 +8292,16 @@ def merge_dicts(*dicts):
            r"""
 remainder(input, other, *, out=None) -> Tensor
 
-Like :func:`torch.fmod` this applies C++'s `std::fmod <https://en.cppreference.com/w/cpp/numeric/math/fmod>`_
-for floating point tensors and the modulus operation for integer tensors.
-Unlike :func:`torch.fmod`, however, if the sign of the modulus is different
-than the sign of the divisor :attr:`other` then the divisor is added to the modulus.
+Computes
+`Python's modulus operation <https://docs.python.org/3/reference/expressions.html#binary-arithmetic-operations>`_
```
Comment (albanD) [inline torch/_torch_docs.py:8296]: Why the newline?

**Review by albanD — COMMENTED** [review]
Sounds good.

**Review by nikitaved — APPROVED** [review]
Forward/Backward AD LGTM!

File: torch/_torch_docs.py (hunk 1 above)
Comment (Lezcano) [reply torch/_torch_docs.py:8296]: The linter was complaining, and I did not want to figure out what's the correct way to split that hyperlink in sphinx :P

Comment (albanD) [reply torch/_torch_docs.py:8296]: ok!

**Review by albanD — APPROVED** [review]
Looks good on my end.
Leaving it to @mruberry for the docs review and landing.

**Comment by soulitzer** [comment]
Updated https://github.com/pytorch/pytorch/issues/71117 to link to this.

**Comment by mruberry** [comment]
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33774417).

**Comment by mruberry** [comment]
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33774417).

**Comment by mruberry** [comment]
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33774417).

**Pull request closed without merging**